### PR TITLE
[win10] added support of xbox media remote control

### PR DESCRIPTION
**Pull request by afedchin**
see title

@koying @Paxxi can you please check if you have a remote

### CODE REVIEW

**Comment by koying** [comment]
Works mostly fine.

What works:
- directional pad
- Ok (Debug Print: CInputManager::HandleKey: 11 (0x0b, obc244) pressed, action is OSD)
- Back (Debug Print: CInputManager::HandleKey: menu (0xd8) pressed, action is Back)
- Play (Debug Print: CInputManager::HandleKey: homepage (0xea) pressed, action is PlayPause) 
- SkipNext (Debug Print: CInputManager::HandleKey: sleep (0xdf) pressed, action is SkipNext)
- SkipPrevious (Debug Print: CInputManager::HandleKey: pause (0xdd) pressed, action is SkipPrevious)

What doesn't (this is via Harmony, so might be a mapping issue there as well):
- Pause (not registered)
- FFwd (not registered)
- Rewind  (not registered)
- Stop (not registered)
- Menu? (Debug Print: CInputManager::HandleKey: epg (0xec) pressed, action is PreviousMenu)

**Comment by afedchin** [comment]
Pause/FFwd/Rewind/Stop will work after handling playback status

**Comment by afedchin** [comment]
@koying please try updated version

File: xbmc/platform/win10/input/RemoteControlXbox.h [hunk 1]
```diff
@@ -20,9 +20,12 @@
  */
 
 #include "input/remote/IRemoteControl.h"
+#include "interfaces/IAnnouncer.h"
+#include "utils/Variant.h"
```
Comment (garbear) [inline xbmc/platform/win10/input/RemoteControlXbox.h]: This header can go in the .cpp file

File: xbmc/platform/win10/input/RemoteControlXbox.h [hunk 2]
```diff
@@ -46,6 +49,8 @@ class CRemoteControlXbox : public KODI::REMOTE::IRemoteControl
 
   static bool IsRemoteControlId(const std::wstring &deviceId);
 
+  void Announce(AnnouncementFlag flag, const char *sender, const char *message, const CVariant &data) override;
```
Comment (koying) [inline xbmc/platform/win10/input/RemoteControlXbox.h]: Doesn't build. Needs to be `ANNOUNCEMENT::AnnouncementFlag`

**Comment by afedchin** [comment]
merged. if there is still an issue we will fix it later.

**Pull request merged**